File: backend/src/main/java/br/com/visitsafe/dto/user/EmployeeUserCreateRequestDTO.java
package br.com.visitsafe.dto.user;

import br.com.visitsafe.model.Account;
import br.com.visitsafe.model.condominium.Condominium;
import br.com.visitsafe.model.enums.DocumentTypeEnum;
import lombok.Data;

@Data
public class EmployeeUserCreateRequestDTO {
    private String name;
    private String rawDocumentNumber;
    private DocumentTypeEnum documentType;
    private Account account;
    private Condominium condominium;
}

File: frontend/src/api/models/EmployeeUserCreateRequestDTO.ts
import type { Account } from './Account.ts'
import type { Condominium } from './Condominium.ts'

export const employeeUserCreateRequestDTODocumentTypeEnum = {
  CPF: 'CPF',
  CNPJ: 'CNPJ',
} as const

export type EmployeeUserCreateRequestDTODocumentTypeEnum =
  (typeof employeeUserCreateRequestDTODocumentTypeEnum)[keyof typeof employeeUserCreateRequestDTODocumentTypeEnum]

export type EmployeeUserCreateRequestDTO = {
  /**
   * @type string | undefined
   */
  name?: string
  /**
   * @type string | undefined
   */
  rawDocumentNumber?: string
  /**
   * @type string | undefined
   */
  documentType?: EmployeeUserCreateRequestDTODocumentTypeEnum
  /**
   * @type object | undefined
   */
  account?: Account
  /**
   * @type object | undefined
   */
  condominium?: Condominium
}